DESIGN.md

```diff
@@ -0,0 +1,370 @@
+# Apollo Android design discussion
+
+These are some preliminary ideas for mapping GraphQL query results to Java, based on what we currently do for Swift in Apollo iOS.
+
+## Query classes
+
+I would expect us to generate classes for queries (and other operations). Instances of these classes represent a particular query, fully configured with the required variables.
```

Comment:
I think we need to know more about how we want to use these query instances before we decide on an architecture, but I agree we want flexibility.